bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -7043,9 +7054,18 @@ public void restoreState(IMemento memento) {
 	@Override
 	public void saveState(IMemento memento) {
 		ISelection selection= doGetSelection();
-		if (selection instanceof ITextSelection) {
-			memento.putInteger(TAG_SELECTION_OFFSET, ((ITextSelection)selection).getOffset());
-			memento.putInteger(TAG_SELECTION_LENGTH, ((ITextSelection)selection).getLength());
+		if (selection instanceof IMultiTextSelection multiSelect && multiSelect.getRegions().length > 1) {
+			memento.putBoolean(TAG_IS_MULTISELECTION, true);
+			memento.putInteger(TAG_MULTISELECTION_COUNT, multiSelect.getRegions().length);
```

Comment:
I don't think those 2 flags are necessary, they're redundant. You can just check the memento to deduce those data.